pkg/actuators/machine/instances.go

```diff
@@ -214,3 +214,14 @@ func getServiceInstanceID(serviceInstanceResource machinev1.PowerVSResource, cli
 		return nil, fmt.Errorf("failed to find an ServiceInstanceID, Unexpected serviceInstanceResource type: %s supports only %s and %s", serviceInstanceResource.Type, machinev1.PowerVSResourceTypeID, machinev1.PowerVSResourceTypeName)
 	}
 }
+
+// vmIP holds the vm name and corresponding dhcp ip used to cache the dhcp ip
+type vmIP struct {
+	name string
+	ip   string
+}
+
+// CacheKeyFunc defines the key function required in TTLStore.
+func CacheKeyFunc(obj interface{}) (string, error) {
+	return obj.(vmIP).name, nil
+}
```

Comment:
wondering if we can move to a different file like caching.go or something like that.